kafka_consumer/tests/test_unit.py

```diff
@@ -515,23 +522,149 @@ def test_deserialize_message():
         b'\x00\x00\x00\x01\x5e{"name": "Peter Parker", "age": 18, "transaction_amount": 123, "currency": "dollar"}'
     )
     key = b'{"name": "Peter Parker"}'
-    assert deserialize_message(MockedMessage(message, key)) == (
+    assert deserialize_message(MockedMessage(message, key), 'json', '', 'json', '') == (
         '{"name": "Peter Parker", "age": 18, "transaction_amount": 123, "currency": "dollar"}',
         None,
         '{"name": "Peter Parker"}',
         None,
     )
-    assert deserialize_message(MockedMessage(message_with_schema)) == (
+    assert deserialize_message(MockedMessage(message_with_schema), 'json', '', 'json', '') == (
         '{"name": "Peter Parker", "age": 18, "transaction_amount": 123, "currency": "dollar"}',
         350,
         '',
         None,
     )
     invalid_json = b'{"name": "Peter Parker", "age": 18, "transaction_amount": 123, "currency": "dollar"'
-    assert deserialize_message(MockedMessage(invalid_json, key)) == (None, None, None, None)
+    assert deserialize_message(MockedMessage(invalid_json, key), 'json', '', 'json', '') == (None, None, None, None)
 
     invalid_utf8 = b'{"name": "Peter Parker", "age": 18, "transaction_amount": 123, "currency": "dollar"\xff'
-    assert deserialize_message(MockedMessage(invalid_utf8, key)) == (None, None, None, None)
+    assert deserialize_message(MockedMessage(invalid_utf8, key), 'json', '', 'json', '') == (None, None, None, None)
+
+    # Test Avro deserialization
+    avro_schema = (
+        '{"type": "record", "name": "Book", "namespace": "com.book", '
+        '"fields": [{"name": "isbn", "type": "long"}, {"name": "title", "type": "string"}, '
+        '{"name": "author", "type": "string"}]}'
+    )
+    avro_message = b'\xd0\xf5\xe4\xd6\xa3\xb9\x046The Go Programming Language\x18Alan Donovan'
+    parsed_avro_schema = build_schema('avro', avro_schema)
+    assert deserialize_message(MockedMessage(avro_message, key), 'avro', parsed_avro_schema, 'json', '') == (
+        '{"isbn": 9780134190440, "title": "The Go Programming Language", "author": "Alan Donovan"}',
+        None,
+        '{"name": "Peter Parker"}',
+        None,
+    )
+
+    # Test Protobuf deserialization
+    protobuf_schema = (
+        'CmoKDHNjaGVtYS5wcm90bxIIY29tLmJvb2siSAoEQm9vaxISCgRpc2JuGAEgASgDUgRpc2Ju'
+        'EhQKBXRpdGxlGAIgASgJUgV0aXRsZRIWCgZhdXRob3IYAyABKAlSBmF1dGhvcmIGcHJvdG8z'
+    )
+    protobuf_message = (
+        b'\x08\xe8\xba\xb2\xeb\xd1\x9c\x02\x12\x1b\x54\x68\x65\x20\x47\x6f\x20\x50\x72\x6f\x67\x72\x61\x6d\x6d\x69\x6e\x67\x20\x4c\x61\x6e\x67\x75\x61\x67\x65'
+        b'\x1a\x0c\x41\x6c\x61\x6e\x20\x44\x6f\x6e\x6f\x76\x61\x6e'
+    )
+    parsed_protobuf_schema = build_schema('protobuf', protobuf_schema)
```

Comment:
The schema is in bytes here but I believe the remote config will have string representation. Will there be an issue with that?